src/amo/components/AddonReview.js

```diff
@@ -19,40 +17,52 @@ export class AddonReviewBase extends React.Component {
     errorHandler: PropTypes.object.isRequired,
     i18n: PropTypes.object.isRequired,
     review: PropTypes.object.isRequired,
-    router: PropTypes.object.isRequired,
     updateReviewText: PropTypes.func,
   }
 
-  onSubmit = (event) => {
+  constructor(props) {
+    super(props);
+    this.state = { reviewTitle: null };
+    if (props.review) {
+      // TODO: also do this in componentWillReceiveProps.
+      this.state.reviewTitle = props.review.title;
+      this.state.reviewBody = props.review.body;
+    }
+    this.overlayCard = null;
+    this.reviewTextarea = null;
+    this.reviewTitleInput = null;
+  }
+
+  onSubmit = (event, { overlayCard = this.overlayCard } = {}) => {
```

Comment:
I dislike abstraction for the sake of abstraction. If two different event handlers needed to call `updateReviewText()` then that would be a good reason to split it into a new function but for now there's only one. Another reason to split it out would be to make testing easier but that's not a problem here.